matchbot-cli.php

```diff
@@ -48,6 +50,11 @@
         $psr11App->get(LoggerInterface::class),
         [$messengerReceiverKey],
     ),
+    new DeleteStalePaymentDetails(
+        new \DateTimeImmutable('now'),
```

Comment:
Liking the date DI

Reply:
(on the other hand in the unit tests you've repeated the same code `new \DateTimeImmutable('now');` which means we aren't actually getting any benefit from DI-ing it. As mentioned below I think the tests could be quite a bit simpler and less like the production code by using a hard-coded date.